src/marqo/tensor_search/utils.py

```diff
@@ -175,20 +176,52 @@ def read_env_vars_and_defaults(var: str) -> Optional[str]:
     """Attempts to read an environment variable.
     If none is found, it will attempt to retrieve it from
     configs.default_env_vars(). If still unsuccessful, None is returned.
+    If it's an empty string, None is returned.
```

Comment:
There is a risk here we can no longer distinguish between 
1. a value was not provided
2. An 'empty' value was provided.

I don't believe right now this is an issue, but it is often required.

Reply:
Yep I agree this is an issue, but this is the functionality implemented originally - worth keeping in mind as future work.